src/Grids/rectilinear_grid.jl

```diff
@@ -1,4 +1,4 @@
-struct RectilinearGrid{FT, TX, TY, TZ, FX, FY, FZ, VX, VY, VZ, Arch} <: AbstractUnderlyingGrid{FT, TX, TY, TZ, Arch}
+struct RectilinearGrid{FT, TX, TY, TZ, FX, FY, FZF, FZC, VX, VY, VZ, Arch} <: AbstractUnderlyingGrid{FT, TX, TY, TZ, Arch}
```

Comment:
Why do FX, FY take priority over FZF and FZC (not to mention Arch) here?

Reply:
I don't think we ever dispatch on Arch. FZF and FZC can definitely take priority. I ll open a new PR to do that.